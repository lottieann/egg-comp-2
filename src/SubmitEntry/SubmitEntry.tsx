import * as React from 'react'

import styled from 'styled-components'

const Container = styled.div`
  margin-top: 15vh;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-content: center;
`

const MessageWrapper = styled.div`
  width: 50vw;

  @media screen and (max-width: 600px) {
    width: 100vw;
  }
`

const Message = styled.p`
  font-family: Helvetica;
  font-size: 16px;
  background-color: rgba(255, 255, 255, 0.7);
  padding: 20px;
  line-height: 150%;
`

const Prerequisites = styled.ul`
  padding-left: 20px;
`

const ListItem = styled.li`
  margin: 0;
`

const Email = styled.p`
  display: flex;
  font-weight: bold;
  font-size: 18px;
  justify-content: center;
`

export const SubmitEntry = () => {
  return (
    <Container data-testid="submit-entry-container">
      <MessageWrapper>
        <Message>
          Please submit your entries by <b>3pm Good Friday</b>, stating who has
          entered which egg, to:
          {'\n'}
          <Email>eastereggcompetition@gmail.com</Email>
          <Prerequisites>
            <ListItem>
              Photos and videos must be <b>landscape</b>
            </ListItem>
            <ListItem>
              Only a total of 3 photos and videos, e.g. 1 video &#38; 2 photos
              OR 3 photos
            </ListItem>
            <ListItem>Repeat entries will be disqualified</ListItem>
          </Prerequisites>
          Voting to take place on Easter Sunday morning. Entries will be
          available for viewing from Saturday Morning. There are 5 votes per
          person.
          {'\n'}
          The winner will be announced <b>12 noon Easter Sunday</b>
        </Message>
      </MessageWrapper>
    </Container>
  )
}
