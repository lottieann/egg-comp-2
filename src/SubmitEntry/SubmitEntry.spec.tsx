import * as React from 'react'

import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'

import { render } from '@testing-library/react'
import { SubmitEntry } from './SubmitEntry'

const setup = () => {
  return <SubmitEntry />
}

describe('SubmitEntry', () => {
  it('should render SubmitEntry correctly', () => {
    const { container } = render(setup())

    expect(container).toMatchSnapshot()
  })
})
