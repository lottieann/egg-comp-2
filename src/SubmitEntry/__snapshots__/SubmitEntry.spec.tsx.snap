// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitEntry should render SubmitEntry correctly 1`] = `
.c0 {
  margin-top: 15vh;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c1 {
  width: 50vw;
}

.c2 {
  font-family: Helvetica;
  font-size: 16px;
  background-color: rgba(255,255,255,0.7);
  padding: 20px;
  line-height: 150%;
  margin: 0;
}

.c4 {
  padding-left: 20px;
}

.c5 {
  margin: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
  font-size: 18px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media screen and (max-width:600px) {
  .c1 {
    width: 100vw;
  }
}

<div>
  <div
    class="c0"
    data-testid="submit-entry-container"
  >
    <div
      class="c1"
    >
      <p
        class="c2"
      >
        Please submit your entries by 
        <b>
          3pm Good Friday
        </b>
        , stating who has entered which egg, to:
        

        <p
          class="c3"
        >
          eastereggcompetition@gmail.com
        </p>
        <ul
          class="c4"
        >
          <li
            class="c5"
          >
            Photos and videos must be 
            <b>
              landscape
            </b>
          </li>
          <li
            class="c5"
          >
            Only a total of 3 photos and videos, e.g. 1 video & 2 photos OR 3 photos
          </li>
          <li
            class="c5"
          >
            Repeat entries will be disqualified
          </li>
        </ul>
        Voting to take place on Easter Sunday morning. Entries will be available for viewing from Saturday Morning. There are 5 votes per person.
        

        The winner will be announced 
        <b>
          12 noon Easter Sunday
        </b>
      </p>
    </div>
  </div>
</div>
`;
